frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -261,8 +243,9 @@ function FileManagerHistory:onShowHist(search_info)
         onLeftButtonTap = function() self:showHistDialog() end,
         onMenuChoice = self.onMenuChoice,
         onMenuHold = self.onMenuHold,
-        onSetRotationMode = self.MenuSetRotationModeHandler,
         _manager = self,
+        _show = self.onShowHist,
+        _args = { search_info },
```

Comment:
Not familiar with anything rotation in KOReader to comment more.
But these 2 params may have more explicite names , and even if private, could  be documented in `Menu = FocusManager:extend{}` - because right now reading this, I have no obvious idea what this means, :)

Reply:
Or just `_caller`. A method that builds and shows the Menu instance.

Reply:
`show_parent` is already used and has some explicite meaning: it's the upper widget to use when inner widgets need to refresh themselves. So, a widget, not a function, I guess you can't re-use it.

May be `_recreate_func` and `_recreate_args` ?

Reply:
Or just a `_recreate_func = function() self:onShowHist(search_info) end` although I don't know how much of the stuff is still valid once the first instance of the widget has been closed.

Reply:
> function() self:onShowHist(search_info) end

Is it the same as `self._show(self._manager)`?
The latter uses the actual `_manager` (i.e. FileManagerHistory instance, that can be changed by a user while viewing the History).

Does the proposed function do the same?
Or it gets the freezed state of `self` at the moment of the Menu instance creation?

Reply:
Dunno, not familiar with what you're doing :) so it would have to be tested.
But I believe anything referenced inside a closure (anonymous function) is still alive (not yet GC) as long as the closure is alive.
I think even with your `_show = self.onShowHist`, you are expecting self and self._manager exist.
Or you can make it even more explicite, using `FileManagerHistory.onShowHist(self._manager, ...)`
(I'm mainly talking about readability here.)

Reply:
Tested, yes, it works.